applications/crossbar/src/modules/cb_cdrs.erl

```diff
@@ -255,465 +231,172 @@ validate(Context, CDRId) ->
-
--spec create_view_options(api_binary(), cb_context:context(), gregorian_seconds(), gregorian_seconds()) ->
-                                 {'ok', crossbar_doc:view_options()}.
-create_view_options('undefined', Context, CreatedFrom, CreatedTo) ->
-    StartKey = maybe_get_start_key_from_req(Context, CreatedTo),
-    {'ok'
-    ,props:filter_undefined(
-       [{'startkey', StartKey}
-       ,{'endkey', CreatedFrom}
-       ,{'limit', pagination_page_size(Context)}
-       ,'descending'
-       ])
+-spec get_view_options(req_nouns()) -> {api_ne_binary(), crossbar_view:options()}.
+get_view_options([{<<"cdrs">>, []}, {?KZ_ACCOUNTS_DB, _}|_]) ->
+    {?CB_LIST, []};
+get_view_options([{<<"cdrs">>, []}, {<<"users">>, [OwnerId]}|_]) ->
+    {?CB_LIST_BY_USER
+    ,[{'start_key_map', [OwnerId]}
+     ,{'end_key_map', [OwnerId]}
+     ]
     };
-create_view_options(OwnerId, Context, CreatedFrom, CreatedTo) ->
-    StartKey = maybe_get_start_key_from_req(Context, [OwnerId, CreatedTo]),
-    {'ok'
-    ,props:filter_undefined(
-       [{'startkey', StartKey}
-       ,{'endkey', [OwnerId, CreatedFrom]}
-       ,{'limit', pagination_page_size(Context)}
-       ,'descending'
-       ])
-    }.
-
--spec create_interaction_view_options(api_binary(), cb_context:context(), pos_integer(), pos_integer()) ->
-                                             {'ok', crossbar_doc:view_options()}.
-create_interaction_view_options('undefined', Context, CreatedFrom, CreatedTo) ->
-    StartKey = maybe_get_start_key_from_req(Context, [CreatedTo]),
-    {'ok'
+get_view_options([{<<"cdrs">>, [?PATH_INTERACTION]}, {?KZ_ACCOUNTS_DB, _}|_]) ->
+    {?CB_INTERACTION_LIST
     ,props:filter_undefined(
-       [{'startkey', StartKey}
-       ,{'endkey', [CreatedFrom, kz_json:new()]}
-       ,{'limit', pagination_page_size(Context)}
+       [{'start_key_map', []}
+       ,{'end_key_map', crossbar_view:suffix_key_fun([kz_json:new()])}
        ,{'group', 'true'}
        ,{'group_level', 2}
        ,{'reduce', 'true'}
-       ,'descending'
         | maybe_add_stale_to_options(?STALE_CDR)
        ])
     };
-create_interaction_view_options(OwnerId, Context, CreatedFrom, CreatedTo) ->
-    StartKey = maybe_get_start_key_from_req(Context, [OwnerId, CreatedTo]),
-    {'ok'
+get_view_options([{<<"cdrs">>, [?PATH_INTERACTION]}, {<<"users">>, [OwnerId]}|_]) ->
+    {?CB_INTERACTION_LIST_BY_USER
     ,props:filter_undefined(
-       [{'startkey', StartKey}
-       ,{'endkey', [OwnerId, CreatedFrom, kz_json:new()]}
-       ,{'limit', pagination_page_size(Context)}
+       [{'start_key_map', [OwnerId]}
+       ,{'end_key_map', fun(Ts) -> [OwnerId, Ts, kz_json:new()] end}
        ,{'group', 'true'}
        ,{'group_level', 3}
        ,{'reduce', 'true'}
-       ,'descending'
         | maybe_add_stale_to_options(?STALE_CDR)
        ])
-    }.
+    };
+get_view_options(_) ->
+    {'undefined', []}.
 
 -spec maybe_add_stale_to_options(crossbar_doc:view_options()) -> crossbar_doc:view_options().
```

Comment:
Can you fix this spec? Should be `boolean()` instead of `crossbar_doc:view_options()`. Let's make the second clause match `false` too instead of `_`.